climateconnect/climateconnect
# Feat: Display similar projects on project page

## Description
Closes #510 

- Changes to dependencies: Added pandas 1.4.3 and updated pytz from version 2019.3 to 2020.1.
- Integrated backend function _(implementation seen here https://github.com/climateconnect/climateconnect/pull/897)_ to get 5 similar projects.
- In the frontend, these 5 similar projects are displayed on a toggle-able sidebar on the right hand side of the screen. At the bottom of the list the user is provided a link which when clicked directs to them /browse. The sidebar is toggle-able via a button on the side of the users screen.  For smaller screen sizes, similar projects are displayed at the bottom of each tab via a existing component `ProjectPreviews`. At the top of the list, the user can minimize the list or display it. At the bottom of the list the user can go directly to /browse and view all existing projects.
- Some buttons e.g. Follow / Contact are adjusted based on whether the sidebar is displayed or screen size.
- `FollowButton.js `refactored to make it more generic


## Test plan

1. Head a page of a project
2. Sidebar should be visible and displayed by default
3. Click the button "hide similar projects" to hide it
4. Click the button "show similar projects" to display it
5. To see small screen size content, reduce screen width to <960px

## Before landing

1. PR has meaningful title
1. `yarn lint` passes (frontend)
1. `yarn format` passes (frontend)
1. `make format` passes (backend)
